### ISSUE TITLE
Different diagnostics in watch vs compile mode

### ISSUE THREAD

**Issue by OmarTawfik** [Microsoft/TypeScript#30640]
**TypeScript Version:**  `3.4.0-dev.20190328`

**Search Terms:** `watch diagnostics` `different` `ts2302` `compile vs watch`

**Code**

```ts
export class MyClass<T> {
  public constructor(public readonly value: T) {}

  public static create(
    // @ts-ignore
    value: T,
    // @ts-ignore
  ): MyClass<T> {
    return new MyClass(value);
  }
}
```

**Expected behavior:**

* Running `tsc file.ts` to compile produces no diagnostics
* Running `tsc file.ts -w` to watch produces no diagnostics

**Actual behavior:**

* Running `tsc file.ts` to compile produces no diagnostics
* Running `tsc file.ts -w` to watch produces the following error:

```
file.ts:4:17 - error TS2302: Static members cannot reference class type parameters.

4   public static create(
                  ~~~~~~
```
Note that:

1) the error (reported on the method name) is extra when running the watcher. The compiler does not report it.
2) when removing the `@ts-ignore`, the compiler reports 2 errors, while the watcher reports 3.

**Comment by RyanCavanaugh** [comment]
In general you are liable for weird results if you use `ts-ignore` in a declaration position  - the compiler is not guaranteed behave consistently if using `ts-ignore` to suppress errors relating to name resolution.

**Comment by OmarTawfik** [comment]
@RyanCavanaugh I understand that diagnostics locations might be off (less accurate), as the compiler might not build a full symbol table when it finds/ignores errors, **but** the `watch` and `build` actions should always produce the same set of diagnostics.

**Comment by RyanCavanaugh** [comment]
We're not going to do the large amount of architectural work required to do that for the sake of allowing what is effectively a syntax error to be ignored in a more ergonomic fashion.
